R/spillover.R

```diff
@@ -191,32 +195,43 @@ setMethod("spillover",
               
               # background correction
               inten <- pmax(sweep(inten[-unstained, ], 2, inten[unstained, ]), 0)
-              
-              # normalize by max of each row
-              inten <- sweep(inten, 1, apply(inten, 1, max), "/")
-              
+
               # Updates the "rownames" of the intensity matrix. If the channel
               # order was not set above, then a guess is made based on the
               # largest statistic for the compensation control (i.e., the row).
               if (any(is.na(channel_order))) {
-                channel_order <- apply(inten, 1, which.max)
-                if (anyDuplicated(channel_order) > 0) {
-                  stop("Unable to match stains with controls based on intensity: ",
-                       "a single stain matches to several multiple controls. ",
-                       call. = FALSE)
+                dists = sapply(colnames(inten),function(x)adist(x,rownames(inten)))
```

Comment:
Taking the edit distance between the `colnames(inten)` and `rownames(inten)`--could be reasonable, but if that's what were doing the above comment is obsolete.